CS-9187 - security checklist update (#1584)
ezsystems/developer-documentation: docs/guide/security_checklist.md

## Underlying stack

> Once you have properly configured secure user roles and permissions, to avoid exposing your application to DDOS vulnerabilities and other, yet unknown security threats, make sure that you do the following:

Reviewer: ```suggestion
Once you have properly configured secure user roles and permissions, to avoid exposing your application to any DDOS vulnerabilities or other yet unknown security threats, make sure that you do the following:
```
Being very clear that there are no _known_ such vulnerabilities.

> - Avoid exposing servers on the open internet when not strictly required.

Reviewer 2: ```suggestion

- Avoid exposing servers on the open internet when not strictly required.
```

- Ensure any servers, services, ports and virtual hosts that were opened for testing purposes are locked down before going live.
- Secure the database with a good password, keys, firewall, etc.

> - Consider whether you must make certain interfaces available on the open internet, for example:

Reviewer: ```suggestion
- Consider whether certain interfaces must be left available on the open internet. Roles protect your content on all interfaces, but you may prefer to reduce your attack surface. For example:
```
...because they are available by default. Also stressing that the services _are_ secure, this is an _optional extra_ level of security.

    - The `/search` and `/graphql` endpoints
    - The REST API endpoints

!!! tip "Access control"

> One way to secure an endpoint that should not be openly available is to restrict access to logged-in users, by using the [`access_control`](https://symfony.com/doc/5.4/security/access_control.html) feature. In your YAML configuration, under the `security` key, add an entry similar to the following one, which redirects requests to a login page:

Reviewer: ```suggestion
One way to lock down an endpoint that should not be openly available is to restrict access to logged-in users, by using the [`access_control`](https://symfony.com/doc/5.4/security/access_control.html) feature. In your YAML configuration, under the `security` key, add an entry similar to the following one, which redirects requests to a login page:
```
The interfaces are built to be secure even when open.

> One way to lock down an endpoint that should not be openly available is to restrict access to logged-in users, by using the [`access_control`](https://symfony.com/doc/5.4/security/access_control.html) feature. In your YAML configuration, under the `security` key, add an entry similar to the following one, which redirects requests to a login page:

Reviewer 2: ```suggestion
  One way to lock down an endpoint that should not be openly available is to restrict access to logged-in users, by using the [`access_control`](https://symfony.com/doc/5.4/security/access_control.html) feature. In your YAML configuration, under the `security` key, add an entry similar to the following one, which redirects requests to a login page:

```yaml
security:
    access_control:
        - { path: ^/search, roles: ROLE_USER}

> ```

Reviewer 2: ```suggestion
```
```
```
